app/controllers/jobseekers/profiles/hide_profile_controller.rb

```diff
@@ -0,0 +1,66 @@
+class Jobseekers::Profiles::HideProfileController < Jobseekers::ProfilesController
+  def show
+    @form = Jobseekers::Profile::HideProfileForm.new(hide_profile: profile.hide_profile)
+  end
+
+  def confirm_hide
+    form_params = params
+      .require(:jobseekers_profile_hide_profile_form)
+      .permit(:hide_profile)
+
+    if (@form = Jobseekers::Profile::HideProfileForm.new(form_params)).valid?
+      profile.update(hide_profile: @form.hide_profile)
```

Comment:
Nitpick: should we use `update!` here instead - non-banged version risks failed save without any feedback